File: leetcode3/변지협/v2/3978. Unique Middle Element.py

'''
1. 아이디어 :
middle 원소 구하고, 그 원소가 유일한지 확인한다.

2. 시간복잡도 :
    o(n)
3. 자료구조/알고리즘 :
'''


class Solution:
    def isMiddleElementUnique(self, nums: list[int]) -> bool:
        n = len(nums)
        middle = nums[n//2]
        return True if 1 == len([i for i in nums if i == middle]) else False

File: leetcode3/염혜정/3978. Unique Middle Element.java
class Solution {
    public boolean isMiddleElementUnique(int[] nums) {
        int middle = (nums.length - 1) / 2;
        int cnt = 0;
        for (int num : nums) {
            if (num == nums[middle]) cnt++;
            if (cnt == 2) break;
        }
        return cnt == 1;
    }
}
